sources/packages/web/src/components/common/ProgramDetails.vue

```diff
@@ -0,0 +1,168 @@
+        >Classification of Instructional Programs (CIP)</span
+      >
+      <br />
+      <p>{{ educationProgram.cipCode }}</p>
+    </v-col>
+    <v-col cols="4"
+      ><span class="font-weight-bold"
+        >National Occupational Classification (NOC)</span
+      >
+      <br />
+      <p>{{ educationProgram.nocCode }}</p>
+    </v-col>
+    <v-col cols="3"
+      ><span class="font-weight-bold">Institution Program Code</span>
+      <br />
+      <p>{{ educationProgram.institutionProgramCode }}</p>
+    </v-col>
+  </v-row>
+</template>
+
+<script lang="ts">
+import { useRouter } from "vue-router";
+import { onMounted, ref, computed } from "vue";
+import {
+  InstitutionRoutesConst,
+  AESTRoutesConst,
+} from "@/constants/routes/RouteConstants";
+import { EducationProgramService } from "@/services/EducationProgramService";
+import { EducationProgramData, ProgramIntensity, ClientIdType } from "@/types";
+import StatusBadge from "@/components/generic/StatusBadge.vue";
+import { useFormatStatuses } from "@/composables";
+
+export default {
+  components: { StatusBadge },
+  props: {
+    programId: {
+      type: Number,
+      required: true,
+    },
+    locationId: {
+      type: Number,
+      required: true,
+    },
+    clientType: {
+      type: String,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const router = useRouter();
+    const { getProgramStatusToGeneralStatus } = useFormatStatuses();
+
+    const isInstitutionUser = computed(() => {
+      return props.clientType === ClientIdType.Institution;
+    });
+
+    const isAESTUser = computed(() => {
+      return props.clientType === ClientIdType.AEST;
+    });
+
+    const programActionLabel = computed(() => {
+      return isInstitutionUser.value ? "Edit" : "View Program";
+    });
+
+    const programButtonAction = () => {
+      if (isInstitutionUser.value) {
+        router.push({
+          name: InstitutionRoutesConst.EDIT_LOCATION_PROGRAMS,
+          params: {
+            programId: props.programId,
+            locationId: props.locationId,
+            clientType: ClientIdType.Institution,
+          },
+        });
+      }
+
+      if (isAESTUser.value) {
+        router.push({
+          name: AESTRoutesConst.VIEW_PROGRAM,
+          params: {
+            programId: props.programId,
+            locationId: props.locationId,
+            clientType: ClientIdType.AEST,
+          },
+        });
+      }
+    };
+
+    const educationProgram = ref({} as EducationProgramData);
+    const getEducationProgramAndOffering = async () => {
+      if (isInstitutionUser.value) {
+        educationProgram.value = await EducationProgramService.shared.getEducationProgram(
+          props.programId,
+        );
+      } else if (isAESTUser.value) {
+        educationProgram.value = await EducationProgramService.shared.getEducationProgramForAEST(
+          props.programId,
+        );
+      }
+    };
+
+    onMounted(getEducationProgramAndOffering);
```

Comment:
await ?

Reply:
There is no need in this context 😉

Reply:
Ah Yes. await is inside getEducationProgramAndOffering.